util/comparator.cc

```diff
@@ -98,6 +98,26 @@ class BytewiseComparatorImpl : public Comparator {
     }
     // *key is a run of 0xffs.  Leave it alone.
   }
+
+  // TODO(Zhongyi): add test case coverage
+  virtual bool IsSameLengthImmediateSuccessor(const Slice& s,
+                                              const Slice& t) const override {
+    if (s.size() != t.size() || s.size() == 0) { return false; }
+    size_t diff_ind = s.difference_offset(t);
+    // should only differ in the last byte
+    // s, t are non empty, safe to check s.size() - 1
+    if (diff_ind != s.size() - 1) {
+      return false;
+    }
+    uint8_t byte_s = static_cast<uint8_t>(s[diff_ind]);
+    uint8_t byte_t = static_cast<uint8_t>(t[diff_ind]);
+    if (byte_s != uint8_t{0xff} && byte_s + 1 == byte_t) {
+      return true;
```

Comment:
I don't understand this. You are only handling last two bytes? If both of the last two bytes are 0xFF, it is not handled?

Reply:
the only case where this function returns true is when 1) two same length Slices only differ in the last byte and 2) the last byte of `s` plus 1 is equal last byte of `t`. 
If they differ in a previous location then `t` cannot be immediate successor of `s`.
For the case of `byte_s=0xff` and `byte_t=0xff` the correct answer is also `false` and that's being handled too.
There is a corner case of s and t both have size of 1, then I assume we don't consider `0xff` and `0x0000` being consecutive?

Reply:
Why not make the implementation clean? Handle case like 0x05FF to 0x0600?